packages/components/src/templates/next/layouts/Collection/utils/__tests__/sortCollectionItems.test.ts

```diff
@@ -0,0 +1,213 @@
+import { beforeAll, describe, expect, it } from "vitest"
+
+import type { SortableCardProps } from "../sortCollectionItems"
+import { sortCollectionItems } from "../sortCollectionItems"
+
+describe("sortCollectionItems", () => {
+  const createItem = ({
+    title,
+    date,
+    variant = "article",
+    url = "",
+  }: {
+    title: string
+    date?: Date
+    variant?: "file" | "link" | "article"
+    url?: string
+  }): SortableCardProps => {
+    return {
+      title,
+      rawDate: date,
+      variant,
+      url,
+      description: "",
+      category: "Others",
+      site: {
+        siteMap: {
+          id: "root",
+          title: "Test Site",
+          summary: "",
+          lastModified: "2024-01-01",
+          permalink: "/",
+          layout: "homepage",
+        },
+        siteName: "Test Site",
+        theme: "isomer-next",
+        logoUrl: "",
+        search: {
+          type: "localSearch",
+          searchUrl: "/search",
+        },
+        navBarItems: [],
+        footerItems: {
+          siteNavItems: [],
+          privacyStatementLink: "/privacy",
+          termsOfUseLink: "/terms",
+        },
+        lastUpdated: "2024-01-01",
+      },
+    } as SortableCardProps
+  }
+
+  it("should sort items by date (newest first)", () => {
+    // Arrange
+    const items = [
+      createItem({ title: "Oldest", date: new Date("2023-01-01") }),
+      createItem({ title: "Newest", date: new Date("2023-12-31") }),
+      createItem({ title: "Middle", date: new Date("2023-06-15") }),
+    ]
+
+    // Act
+    const sorted = sortCollectionItems({ items })
+
+    // Assert
+    const expectedTitles = ["Newest", "Middle", "Oldest"]
+    expect(sorted.map((item) => item.title)).toEqual(expectedTitles)
+  })
+
+  it("should sort by title when dates are equal", () => {
+    // Arrange
+    const sameDate = new Date("2023-01-01")
+    const items = [
+      createItem({ title: "Charlie", date: sameDate }),
+      createItem({ title: "Alice", date: sameDate }),
+      createItem({ title: "Bob", date: sameDate }),
+    ]
+
+    // Act
+    const sorted = sortCollectionItems({ items })
+
+    // Assert
+    const expectedTitles = ["Alice", "Bob", "Charlie"]
+    expect(sorted.map((item) => item.title)).toEqual(expectedTitles)
+  })
+
+  it("should place items without dates at the end", () => {
```

Comment:
how did we arrive at this btw? Why not at the front or add a default new Date()?

Reply:
title should be "should place items without dates at the end **sorted alphabetically**"

Reply:
> how did we arrive at this btw? Why not at the front or add a default new Date()?

this is an existing behaviour actually

if i recall correctly, the reason why we did this for the edge case of migrated collections that contains both date-variant and non-date variant, and non-date variant are placed at the end partially for aesthetic reasons _(otherwise it's looks jank to have them interweaved)_. This edge case will not be possible for collections created on Studio